security/cmd/istio_ca/main.go

```diff
@@ -143,8 +144,11 @@ func fatalf(template string, args ...interface{}) {
 func init() {
 	flags := rootCmd.Flags()
 	// General configuration.
-	flags.StringVar(&opts.listenedNamespace, "listened-namespace", "",
-		"Select a namespace for the CA to listen to. If unspecified, Citadel tries to use the ${"+
+	flags.StringVar(&opts.listenedNamespaces, "listened-namespace", "", "deprecated")
+	flags.MarkDeprecated("listened-namespace", "please use --listened-namespaces instead")
```

Comment:
I'm a bit of concerned about the flag deprecation policy. Not sure the usage of this flag to our users. This will be a breaking change if customers are depending on this.

@myidpt thoughts?

Reply:
@incfly the deprecated flag will continue to function but will not show up in help or usage messages. Using this flag will also print the given usageMessage. I am not sure when to delete the deprecated flag. What is the flag deprecation policy?

Reply:
I solicit around a bit. Deprecation flag is okay. Can you search to see if there's any reference of this flag from helm values.yaml and replace them as well?

Reply:
@incfly there is no any references except for `/security/cmd/istio_ca/main.go`.

Reply:
how is user going to consume this flag?

Reply:
specify the parameter as args for citadel:
```
   containers:
      - args:
        - --append-dns-names=true
        - --grpc-port=8060
        - --grpc-hostname=citadel
        - --citadel-storage-namespace=istio-system
        - --custom-dns-names=istio-pilot-service-account.istio-system:istio-pilot.istio-system
        - --monitoring-port=15014
        - --listened-namespaces=istio-system,default
        - --self-signed-ca=true
        image: docker.io/istio/citadel:1.1.0-rc.0
```